test/cli/test_log_runtime_archive.rb

```diff
@@ -510,6 +510,102 @@ def should_archive_dataset(dataset, archive_basename, full:)
                 end
             end
 
+            describe "#ensure_free_space" do
+                before do
+                    @archive_dir = make_tmppath
+                    @free_space_low_limit = 1
+                    @free_space_delete_until = 10
+                    @deleted_files = []
+                    @mocked_files_sizes = []
+
+                    10.times { |i| (@archive_dir / i.to_s).write(i.to_s) }
+
+                    @archiver = LogRuntimeArchive.new(
+                        @root, @archive_dir,
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                end
+
+                it "does nothing if there is enough free space" do
+                    mock_available_space(2)
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                    assert_deleted_files([])
+                end
+
+                it "removes enough files to reach the freed limit" do
+                    size_files = [6, 2, 4, 6, 7, 10, 3, 5, 8, 9]
+                    mock_files_size(size_files)
+                    mock_available_space(size_files.sum + 0.5)
+
+                    @archiver.ensure_free_space(
+                        free_space_low_limit: @free_space_low_limit,
+                        free_space_delete_until: @free_space_delete_until
+                    )
+                    assert_deleted_files(@deleted_files)
+                end
+
+                def mock_files_size(sizes)
+                    @mocked_files_sizes = sizes
+                    sizes.each_with_index do |size, i|
+                        flexmock(LogRuntimeArchive)
+                            .should_receive(:size_of_file)
+                            .with(@archive_dir / i.to_s)
+                            .and_return(size)
+                    end
+                end
+
+                def mock_available_space(total_disk_size)
+                    flexmock(Sys::Filesystem)
+                        .should_receive(:stat).with(@archive_dir)
+                        .and_return do
+                            flexmock(
+                                bytes_free: compute_mocked_disk_size(total_disk_size)
+                            )
+                        end
+                end
+
+                def compute_mocked_disk_size(total_disk_size)
+                    return total_disk_size if @mocked_files_sizes.empty?
+
+                    @mocked_files_sizes.each_index do |i|
+                        if !(@archive_dir / i.to_s).exist?
+                            total_disk_size -= @mocked_files_sizes[i]
+                            @deleted_files.append(i)
```

Comment:
You are taking all the files that are actually deleted, not the ones that are supposed to be deleted. Instead of computing the deleted files in here, you should just have a list of the ones that you expect to delete for each unit test.

Reply:
> You are taking all the files that are actually deleted, not the ones that are supposed to be deleted.

Actually that's the main idea of this function... to mock the avaiable space in folder every time a file is deleted. So I do need to compare with files that are deleted.

>Instead of computing the deleted files in here, you should just have a list of the ones that you expect to delete for each unit test.

 I did this in `assert_deleted_files` function.

Reply:
Except that you call `assert_deleted_files` passing the @deleted_files variable, which is the list of files that were deleted, not the list of files that you *expect* to be deleted.

See the dataflow of the variable, it is filled with all files that no longer exist by checking the `@archive_dir / i` path. Then, on assert_deleted_files, you check if the files that were deleted still exist (spoiler: they dont because you checked that to know if they were deleted), or that the files that werent deleted are still there.

Reply:
fixed